src/wp-includes/customize/class-wp-customize-custom-css-setting.php

```diff
@@ -163,7 +163,15 @@ public function validate( $value ) {
 
 		$validity = new WP_Error();
 
-		if ( preg_match( '#</?\w+#', $css ) ) {
+		/**
+		 * Check for a closing STYLE tag inside the CSS.
+		 *
+		 * STYLE tags are processed using the "generic raw text parsing algorithm." They contain
+		 * raw text up until a matching closing tag.
+		 *
+		 * @see https://html.spec.whatwg.org/multipage/parsing.html#generic-raw-text-element-parsing-algorithm
+		 */
+		if ( preg_match( '#</style[ \\t\\f\\n\\r/>]#', $css ) ) {
```

Comment:
Later, the CSS is concatenated and may have newlines added around it, so exclude the trailing character from this check and just look for something problematic-ish:

```suggestion
		if ( stripos( $css, '</style' ) ) {
```